src/diffusers/training_utils.py

```diff
@@ -53,6 +54,49 @@ def compute_snr(noise_scheduler, timesteps):
     return snr
 
 
+def compute_dream_and_update_latents(
```

Comment:
Better would be to return the expected variables rather than doing it in-place. It's more readable that way in a long training script.

Reply:
Sure.